src/_pytest/assertion/util.py

```diff
@@ -319,6 +329,41 @@ def _compare_eq_dict(left, right, verbose=False):
     return explanation
 
 
+def _compare_eq_class(left, right, verbose, type=None):
+    if type == "data":
+        all_fields = left.__dataclass_fields__
+        fields_to_check = [field for field, info in all_fields.items() if info.compare]
+    elif type == "attrs":
+        all_fields = left.__attrs_attrs__
+        fields_to_check = [field.name for field in all_fields if field.cmp]
+    else:
+        raise RuntimeError("Unexpected value for `type` paramater")
+
+    same = []
+    diff = []
+    for field in fields_to_check:
+        if getattr(left, field) == getattr(right, field):
+            same.append(field)
+        else:
+            diff.append(field)
+
+    explanation = []
+    if same and verbose < 2:
+        explanation += [u("Omitting %s identical items, use -vv to show") % len(same)]
+    elif same:
+        explanation += [u("Common attributes:")]
+        explanation += pprint.pformat(same).splitlines()
+    if diff:
+        class_name = left.__class__.__name__
+        explanation += [("Differing attributes:")]
+        for k in diff:
+            explanation += [
```

Comment:
i disagree with the chosen display style but cant provide any meaningful alternative yet - i would like to see some kind of diff style display using multiple lines